tensorflow/python/training/saver.py

```diff
@@ -126,13 +126,10 @@ def _read_variable_closure(v):
           def f():
             with ops.device(v.device):
               x = v.read_value()
-            # To allow variables placed on non-CPU devices to be checkpointed,
-            # we copy them to CPU on the same machine first.
-            device_spec = pydev.DeviceSpec().parse_from_string(v.device)
-            device_spec.merge_from(
-                pydev.DeviceSpec().parse_from_string("/device:CPU:0"))
-            with ops.device(device_spec.to_string()):
-              return array_ops.identity(x)
+              # To allow variables placed on non-CPU devices to be checkpointed,
+              # we copy them to CPU on the same machine first.
+              with ops.device("/device:CPU:0"):
+                return array_ops.identity(x)
```

Comment:
This seems to undo the intent of the original code, because it doesn't ensure that it is copied to CPU *on the same machine*.

Reply:
Agreed. I am not sure why we are reverting the fix.

Reply:
@agarwal-ashish It does the same job as your fix. Because it is under the context of v.device, it will just replace the last part of device to 'device:CPU:0'. 
e.g.
'/job:ps/task:1/device:CPU:1' -->'/job:ps/task:1/device:CPU:0' 
'/job:ps/task:1/device:GPU:0' -->'/job:ps/task:1/device:CPU:0'

Reply:
Ah, that's correct... I missed the indentation change.

Reply:
I missed that too and hence bounced it to the authors. Thanks for pointing out, @jackonan